src/blockdev.rs

```diff
@@ -1320,6 +1331,15 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_gpt_disk_not_flagged_as_mbr() {
```

Comment:
Hmm, can you elaborate on what this is testing for?  It doesn't write a protective MBR, which we would do in real life (in `overwrite`).  And it doesn't actually matter what the function does in isolation, since we only call it if we've found no GPT.

I think instead `test_saved_partitions` should have a section that verifies that we bail if there's an MBR but no GPT.  `GPT::write_protective_mbr_into` might be a good way to construct this.

Reply:
Yeah you are right, It was a misunderstanding on what the new_from did, I assumed(sigh i need to not do that) that the new_from was doing more then just creating a gpt header

Reply:
And yeah that makes sense, I will.